### DISCUSSION TITLE
offline option usable to other apps

### DISCUSSION THREAD

**Discussion by claudio4j** [jbangdev/jbang#1718] [General]
I see there is `--offline` which sets the boolean flag in `dev.jbang.util.Util.offline` field, which works great for `jbang run <my app>` to download the dependencies for my app.
But how can I verify that option in my own app to see if the user is offline ?

**Comment by maxandersen** [comment]
hmm - that info is not really intended to be viewable to the user app as the --offline part is for when building ...not running.

What is your usecase?

**Reply by claudio4j** [reply]
In camel-jbang there is an option to [download dependencies](https://camel.apache.org/manual/camel-jbang.html#_downloading_jars_over_the_internet) as `--download=<true|false>`

Learning about the jbang `--offline` I thought it could be exposed to camel-jbang.

**Reply by maxandersen** [reply]
Here is the thing the app camel jbang on this case runs as separate process thus we can't let the app know except for environment variables. 

But I don't like the idea we set a bunch of env vars for all apps to let you read the used jbang config. 

Ideas welcome. 

One relevant idea is notion jbang plugins where we treat jbang-* found in path or catalogs as commands. Then we know it wants the config and can pass it in. 

would that work?

**Reply by claudio4j** [reply]
> where we treat jbang-* found in path or catalogs as commands. Then we know it wants the config and can pass it in.

That would work and make the jbang apps can look at those configurations.
